### PR TITLE
chore: Make invoice subscription_id nullable

### PR DESCRIPTION
**Pull request by azhur**
## Description
Make invoice subscription_id nullable to be able to create one-off invoices which are not subscription related.

## Checklist
- [ ] The code follows the project's coding conventions and style [guidelines](https://doc.rust-lang.org/1.0.0/style/README.html).
- [ ] All tests related to the changes have passed successfully.
- [ ] Documentation has been updated to reflect the changes (if applicable).
- [ ] All new and existing unit tests have passed.
- [ ] I have self-reviewed my code and ensured its quality.
- [ ] I have added/updated necessary comments to aid understanding.

### CODE REVIEW

File: modules/meteroid/crates/diesel-models/src/query/customers.rs [hunk 1]
```diff
@@ -154,6 +155,53 @@ impl CustomerRow {
             .attach_printable("Error while inserting customer batch")
             .into_db_result()
     }
+
+    pub async fn select_for_update(conn: &mut PgConn, id: Uuid) -> DbResult<CustomerRow> {
```
Comment (azhur) [inline modules/meteroid/crates/diesel-models/src/query/customers.rs:159]: left-over: will be needed for customer wallet feat

File: modules/meteroid/crates/diesel-models/src/query/customers.rs [hunk 2]
```diff
@@ -154,6 +155,53 @@ impl CustomerRow {
...
+        log::debug!("{}", debug_query::<diesel::pg::Pg, _>(&query).to_string());
+
+        query
+            .first(conn)
+            .await
+            .attach_printable("Error while selecting for update customer by id")
+            .into_db_result()
+    }
+
+    pub async fn update_balance(conn: &mut PgConn, id: Uuid, delta_cents: i32) -> DbResult<()> {
```
Comment (azhur) [inline modules/meteroid/crates/diesel-models/src/query/customers.rs:174]: left-over: will be needed for customer wallet feat

File: modules/meteroid/proto/api/invoices/v1/models.proto [hunk 3]
```diff
@@ -23,7 +23,7 @@ message Invoice {
   string invoice_date = 4;
   string customer_id = 5;
   string customer_name = 6;
-  string subscription_id = 7;
+  optional string subscription_id = 7;
```
Comment (azhur) [inline modules/meteroid/proto/api/invoices/v1/models.proto:26]: @gaspb not sure if those changes break ui

Comment (gaspb) [reply modules/meteroid/proto/api/invoices/v1/models.proto:26]: np i'm working on this view :)

File: modules/meteroid/crates/diesel-models/src/query/customers.rs [hunk 4]
```diff
@@ -154,6 +155,53 @@ impl CustomerRow {
...
+        use crate::schema::customer::dsl as c_dsl;
+        use diesel_async::RunQueryDsl;
+
+        let query = diesel::update(c_dsl::customer)
+            .filter(c_dsl::id.eq(id))
+            .filter(c_dsl::balance_value_cents.add(delta_cents).le(0))
```
Comment (gaspb) [inline modules/meteroid/crates/diesel-models/src/query/customers.rs:180]: I guess we want to filter only if >=0 so `.ge` no ?

**Review by gaspb — APPROVED** [review]

File: modules/meteroid/crates/diesel-models/src/query/customers.rs (hunk 4 above)
Comment (azhur) [reply modules/meteroid/crates/diesel-models/src/query/customers.rs]: yep, you are right, fixed, thanks!

**Pull request merged**